### PR TITLE
Generates cos image validation tests

### PR DESCRIPTION
**Pull request by yguo0905**
- The `cosbeta` tests are commented out because we don't want to affect the current running tests at https://k8s-testgrid.appspot.com/cos-image-validation. I will add make the change later in a separate PR.
- The changes for ubuntu tests are because of the changes in k8s channel definition.
- The tests are expected to fail until I increased the quota in the `cos-image-validation` project.

/assign @krzyzacy 
/cc @abgworrall

### CODE REVIEW

File: jobs/ci-kubernetes-e2e-gce-cosdev-k8sbeta-default.env [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# AUTO-GENERATED - DO NOT EDIT.
```
Comment (abgworrall) [inline jobs/ci-kubernetes-e2e-gce-cosdev-k8sbeta-default.env]: No need to fix as part of this commit, but a pet peeve of mine is seeing headers like this that don't tell you what script actually does the generation - it drives me batty when I'm digging into code ! Consider hardcoding the script name in here ...

```
# AUTO-GENERATED by my-awesome-script.py - DO NOT EDIT
```

**Comment by krzyzacy** [comment]
cc @wonderfly 

lg, please rebase, and assume you'll need a lot more quota in your project.

File: jobs/ci-kubernetes-e2e-gce-cosdev-k8sbeta-default.env (hunk 1 above)
Comment (yguo0905) [inline jobs/ci-kubernetes-e2e-gce-cosdev-k8sbeta-default.env]: Makes sense. Will do. Thanks!

**Comment by yguo0905** [comment]
Rebased. PTAL.

**Comment by zmerlynn** [comment]
I'm not sure if you tested this, but this PR generates GKE jobs that use KUBE_GCE_NODE_IMAGE, which is valid in GCE but isn't consumed in GKE. I'll file a bug.

**Comment by yguo0905** [comment]
Yes, I'm aware of this. The plan is to handle cloud provider specific envs in `kubernetes_e2e.py` as mentioned in https://github.com/kubernetes/test-infra/pull/3212#issue-238693619.

I thought this was fine in the current migration process. Is this causing any actual issues?

**Comment by zmerlynn** [comment]
No, it's actually just a nuisance to reasoning about what env variables are currently in use across the GKE tests.

**Comment by yguo0905** [comment]
I've created https://github.com/kubernetes/test-infra/issues/3320 to track this issue. Thanks!

**Pull request merged**